src/components/TodoModal/TodoModal.tsx

```diff
@@ -1,39 +1,67 @@
-import React from 'react';
+import React, { SetStateAction, useEffect, useState } from 'react';
+import { getUser } from '../../api';
+import { Todo } from '../../types/Todo';
+import { User } from '../../types/User';
 import { Loader } from '../Loader';
 
-export const TodoModal: React.FC = () => {
+type Props = {
+  selectedTodo: Todo;
+  setSelectedTodo: React.Dispatch<SetStateAction<Todo | null>>;
+};
+
+export const TodoModal: React.FC<Props> = ({
+  selectedTodo,
+  setSelectedTodo,
+}) => {
+  const [user, setUser] = useState<User>();
+  const [loadingUser, setLoadingUser] = useState<boolean>(false);
+
+  useEffect(() => {
+    setLoadingUser(true);
+
+    getUser(selectedTodo.userId)
+      .then(setUser)
+      .finally(() => setLoadingUser(false));
+  }, [selectedTodo.userId]);
+
   return (
     <div className="modal is-active" data-cy="modal">
       <div className="modal-background" />
+      {loadingUser && <Loader />}
 
-      {true ? (
-        <Loader />
-      ) : (
+      {!loadingUser && (
         <div className="modal-card">
           <header className="modal-card-head">
             <div
               className="modal-card-title has-text-weight-medium"
               data-cy="modal-header"
             >
-              Todo #2
+              Todo #{selectedTodo.id}
             </div>
 
-            {/* eslint-disable-next-line jsx-a11y/control-has-associated-label */}
-            <button type="button" className="delete" data-cy="modal-close" />
+            <button
+              type="button"
+              className="delete"
```

Comment:
You are generating a key on render with the 'delete' class. This is not recommended as React uses keys to identify elements and can lead to unexpected behavior.